apps/src/templates/VersionRow.jsx

```diff
@@ -22,57 +26,72 @@ export default class VersionRow extends React.Component {
   }
 
   render() {
-    let button;
+    let buttons = [];
     if (this.props.isLatest) {
-      button = (
+      // this is the placeholder for the latest version (can't be resolved)
+      buttons.push(
         <button
+          key={buttons.length}
           type="button"
           className="btn-default"
           disabled="disabled"
           style={{cursor: 'default'}}
         >
-          {msg.currentVersion()}
+          {msg.latestVersion()}
         </button>
       );
-    } else {
-      button = [
+    } else if (this.props.isProjectOwned) {
+      buttons.push(
+        <button
+          key={buttons.length}
+          type="button"
+          className="btn-info"
+          onClick={this.props.onChoose}
+        >
+          {msg.restoreThisVersion()}
+        </button>
+      );
+    }
+
+    if (!this.props.isViewingVersion) {
+      const user_id = queryParams('user_id');
+      const viewAs = queryParams('viewAs');
+      buttons.push(
         <a
-          key={0}
+          key={buttons.length}
           href={
             location.origin +
             location.pathname +
-            '?version=' +
-            this.props.versionId
+            (this.props.isLatest ? '' : '?version=' + this.props.versionId) +
```

Comment:
Extracting this into a method would make it more readable

Reply:
Done!